server/src/test/java/org/elasticsearch/search/aggregations/bucket/missing/MissingAggregatorTests.java

```diff
@@ -28,81 +28,115 @@
+            builder -> {},
+            (index, doc) -> doc.add(new SortedNumericDocValuesField(fieldType.name(), randomLong())),
             internalMissing -> {
                 assertEquals(internalMissing.getDocCount(), 0);
                 assertFalse(AggregationInspectionHelper.hasValue(internalMissing));
-            });
+            },
+            singleton(fieldType));
     }
 
     public void testMatchAllDocs() throws IOException {
         int numDocs = randomIntBetween(10, 200);
+
+        final MappedFieldType fieldType = new NumberFieldMapper.Builder("_name", NumberType.LONG).fieldType();
+        fieldType.setName("field");
+        final MappedFieldType anotherFieldType = new NumberFieldMapper.Builder("_name", NumberType.LONG).fieldType();
+        anotherFieldType.setName("another_field");
+
         testBothCases(numDocs,
-            "field",
+            fieldType.name(),
             Queries.newMatchAllQuery(),
-            doc -> doc.add(new SortedNumericDocValuesField("another_field", randomLong())),
+            builder -> {},
+            (index, doc) -> doc.add(new SortedNumericDocValuesField(anotherFieldType.name(), randomLong())),
             internalMissing -> {
                 assertEquals(internalMissing.getDocCount(), numDocs);
                 assertTrue(AggregationInspectionHelper.hasValue(internalMissing));
-            });
+            },
+            List.of(fieldType, anotherFieldType));
     }
 
     public void testMatchSparse() throws IOException {
         int numDocs = randomIntBetween(100, 200);
         final AtomicInteger count = new AtomicInteger();
+
+        final MappedFieldType fieldType = new NumberFieldMapper.Builder("_name", NumberType.LONG).fieldType();
+        fieldType.setName("field");
+        final MappedFieldType anotherFieldType = new NumberFieldMapper.Builder("_name", NumberType.LONG).fieldType();
+        anotherFieldType.setName("another_field");
+
         testBothCases(numDocs,
-            "field",
+            fieldType.name(),
             Queries.newMatchAllQuery(),
-            doc -> {
+            builder -> {},
+            (index, doc) -> {
                 if (randomBoolean()) {
-                    doc.add(new SortedNumericDocValuesField("another_field", randomLong()));
+                    doc.add(new SortedNumericDocValuesField(anotherFieldType.name(), randomLong()));
                     count.incrementAndGet();
                 } else {
-                    doc.add(new SortedNumericDocValuesField("field", randomLong()));
+                    doc.add(new SortedNumericDocValuesField(fieldType.name(), randomLong()));
                 }
             },
             internalMissing -> {
                 assertEquals(internalMissing.getDocCount(), count.get());
                 count.set(0);
                 assertTrue(AggregationInspectionHelper.hasValue(internalMissing));
-            });
+            },
+            List.of(fieldType, anotherFieldType));
```

Comment:
Note: in some of the aggregator tests similar to this one (where there's a second field we expect to be absent sometimes), we don't pass a field type for the second field like above. If the test is missing the second field type, that field can't be read even if an aggregator was erroneously trying to collect it

Reply:
We should fix that, could you open a ticket so we don't forget please?

Reply:
I opened https://github.com/elastic/elasticsearch/issues/53250